sky/skylet/providers/ibm/node_provider.py

```diff
@@ -0,0 +1,924 @@
+
+            if stopped_nodes:
+                cli_logger.print(
+                    f"Reusing nodes {stopped_nodes_ids}. "
+                    "To disable reuse, set `cache_stopped_nodes: False` "
+                    "under `provider` in the cluster configuration."
+                )
+
+            for node in stopped_nodes:
+                logger.info(f"Starting instance {node['id']}")
+                self.ibm_vpc_client.create_instance_action(node["id"], "start")
+
+            time.sleep(1)
+
+            for node_id in stopped_nodes_ids:
+                self.set_node_tags(node_id, tags)
+                with self.lock:
+                    if node_id in self.deleted_nodes:
+                        self.deleted_nodes.remove(node_id)
+
+            count -= len(stopped_nodes_ids)
+
+        created_nodes_dict = {}
+
+        # create multiple instances concurrently
+        if count:
+            with cf.ThreadPoolExecutor(count) as ex:
+                for _ in range(count):
+                    futures.append(
+                        ex.submit(self._create_node, base_config, tags))
+
+            for future in cf.as_completed(futures):
+                created_node = future.result()
+                created_nodes_dict.update(created_node)
+
+        all_created_nodes = stopped_nodes_dict
+        all_created_nodes.update(created_nodes_dict)
+
+        # this sleep is required due to race condition with non_terminated_nodes
+        # called in separate thread by autoscaler. not a lost as anyway the vsi
+        # operating system takes time to start. can be removed after
+        # https://github.com/ray-project/ray/issues/28150 resolved
+        time.sleep(5)
+
+        return all_created_nodes
+
+    def _delete_node(self, node_id):
+        """deletes specified instance.
+        if it's a head node delete its IPs if it was created by Ray.
+        updates caches. """
+
+        logger.info("Deleting VM instance {}".format(node_id))
+        node = self._get_cached_node(node_id)
+
+        # get vpc_id to delete if deleting head node which isn't a failed node
+        # pylint: disable=line-too-long
+        vpc_id_to_delete = None
+        if self._get_node_type(node["name"]) == NODE_KIND_HEAD and node["status"] != "failed":
+            vpc_id_to_delete = self.ibm_vpc_client.get_instance(node_id).get_result()["vpc"]["id"]
+
+            # if the request is executed by the remote head like in `sky autostop --down`
+            # , in contrast to `sky down`, use cloud functions to delete the VPC
+            # with its associated resources. 
+            hostname = socket.gethostname()  # returns the instance's (VSI) name
+            if self._get_node_type(hostname) == NODE_KIND_HEAD:
+                self.vpc_provider.remote_cluster_removal(vpc_id_to_delete, self.region)
+
+        try:
+            floating_ips = []
+
+            # get a node's floating ip
+            try:
+                node = self._get_node(node_id)
```

Comment:
How about we use `_get_node` in line 788 and get rid of this line?

Reply:
indeed. i caught it too, and made a few additional changes. no idea why the extra api call was there in the first place.